web-api/src/Kerbero.Infrastructure/Common/Helpers/NukiRestApiClient.cs

```diff
@@ -0,0 +1,181 @@
+using System.Net;
+using FluentResults;
+using Flurl;
+using Flurl.Http;
+using Kerbero.Domain.Common.Errors;
+using Kerbero.Domain.Common.Models.ExternalResponses;
+using Kerbero.Domain.SmartLocks.Errors;
+using Kerbero.Infrastructure.NukiCredentials.Dtos;
+using Kerbero.Infrastructure.SmartLocks.Models;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Kerbero.Infrastructure.Common.Helpers;
+
+public class NukiRestApiClient
+{
+  private readonly ILogger<NukiRestApiClient> _logger;
+  private readonly IConfiguration _configuration;
+  private string _token;
```

Comment:
We doesn't want nullability problems, so it is better pass _token as parameters. As such, you should solve the problem of a developer who not sets the token. (rename in ApiToken)